tiledb/sm/query/reader.cc

```diff
@@ -345,33 +345,43 @@ Status Reader::set_buffer(
     return LOG_STATUS(
         Status::ReaderError("Cannot set buffer; Array schema not set"));
 
+  // Name cannot be null
+  if (attribute == nullptr)
+    return LOG_STATUS(
+        Status::WriterError("Cannot set buffer; Null attribute name"));
+
   // Check that attribute exists
-  if (attribute == nullptr || (attribute != constants::coords &&
-                               array_schema_->attribute(attribute) == nullptr))
+  auto attr_in_schema = array_schema_->attribute(attribute);
+  if (attribute == nullptr ||
+      (attribute != constants::coords && attr_in_schema == nullptr))
     return LOG_STATUS(
         Status::WriterError("Cannot set buffer; Invalid attribute"));
 
+  // Get name of attribute in schema
+  std::string attribute_name =
+      attr_in_schema == nullptr ? constants::coords : attr_in_schema->name();
+
   // Check that attribute is fixed-sized
   bool var_size =
-      (attribute != constants::coords && array_schema_->var_size(attribute));
+      (attribute != constants::coords && attr_in_schema->var_size());
```

Comment:
Since the attribute name is properly retrieved above, `bool var_size = array_shcema_->var_size(attribute_name)` should be equivalent here.